edsl.md

```diff
@@ -365,4 +374,18 @@ If the data is at most 31 bytes long, it is stored in the higher-order bytes (le
 
 ```k
 endmodule
+
+module BUFSTRICT-SYNTAX
+    imports EVM
+
+    syntax ByteArray ::= #bufStrict ( Int , Int ) [function]
+
+endmodule
+
+module BUFSTRICT-HASKELL [symbolic, kore]
+    imports BUFSTRICT-SYNTAX
+
+    rule #bufStrict(_, _) => #Bottom              [owise]
+
+endmodule
 ```
```

Comment:
I want to make new modules `BUF-SYNTAX` and `BUF-HASKELL`, and `BUF-JAVA` here, and to have `EDSL` import `BUF-HASKELL` and `BUF-JAVA`. All declarations and rules for `#buf` and `#bufStrict` will go here. Also add the documentation (markdown style, not inline comments) to those modules explaining the proper way to use them, and how the backends expcet them to be used? (eg. do I use #buf on LHS or RHS of rule? Sholud I use #bufStrict on RHS of claim? etc...)